js/components/SocialShare/index.js

```diff
@@ -0,0 +1,58 @@
+/* global require */
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+import Share from 'react-native-share';
+
+export default class SocialShare extends Component {
+  constructor(props) {
+    super(props);
+
+    this.share = this.share.bind(this);
+  }
+
+  share() {
+    let shareOptions = this.props.itemId ? 
+      this.fromStore() : 
+      Object.assign({}, this.props);
```

Comment:
I was thinking on passing an item id from the store, and the component know how to extract the information.

But I'm going back to my original thought about just passing properties.